src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -263,8 +264,19 @@ bool layout_optimizer::can_fuse_reorder(program_node& prev, program_node& next,
         ((fmt_prev == format::b_fs_yx_fsv4 && fmt_next == format::bfyx) && (prev_dt == data_types::u8 || prev_dt == data_types::i8))))
         return true;
 
-    if (next.is_type<eltwise>() && prev_simple && next_simple)
+    if (next.is_type<eltwise>() && prev_simple && next_simple) {
+        // Accumulate type of eltwise becomes fp32 so that it prevents accuracy degradation.
+        if (data_type_traits::is_floating_point(prev_dt) && data_type_traits::is_floating_point(next_dt) &&
+            (data_type_traits::size_of(prev_dt) < data_type_traits::size_of(next_dt))) {
+            return false;
+        }
         return true;
+    }
+
+    if ((next.is_type<permute>() || next.is_type<fully_connected>())
+        && (fmt_prev == fmt_next) && (prev_dt != next_dt)) {
+        return true;
+    }
```

Comment:
Is this condition required `(prev_dt != next_dt)`? I think it can be fused even if `prev_dt == next_dt`...
By the way, could you add a comment to express the target pattern here?

Reply:
I think you are right. I will remove.